src/Microsoft.ML.StandardLearners/Standard/SdcaMultiClass.cs

```diff
@@ -197,6 +197,8 @@ protected override void CheckLabelCompatible(SchemaShape.Column labelCol)
                         // Loop trials for compare-and-swap updates of duals.
                         // In general, concurrent update conflict to the same dual variable is rare
                         // if data is shuffled.
+                        var weightsEditor = VBufferEditor.CreateFromBuffer(ref weights[iClass]);
+                        var l1IntermediateWeightsEditor = VBufferEditor.CreateFromBuffer(ref l1IntermediateWeights[iClass]);
```

Comment:
>l1IntermediateWeights [](start = 93, length = 21)

I think maybe this is null in the case where the L1 threshold is zero (see condition about 30 lines down on `l1ThresholdZero`) which could explain the test failures you are seeing?

Reply:
Good call. I fixed it with https://github.com/dotnet/machinelearning/pull/1650/commits/327a9c994d147ec06487c91ccd356b74e1f9b64e